lib/fakturoid/oauth/access_token_service.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Fakturoid
+  class Oauth
+    class AccessTokenService
+      attr_reader :oauth, :client
+
+      def initialize(oauth)
+        @oauth  = oauth
+        @client = oauth.client
+      end
+
+      def perform_request(method, path, params)
+        check_access_token
+        fetch_access_token if client.config.credentials.access_token_near_expiration?
+
+        retried = false
+
+        begin
+          Request::Api.new(method, path, client).call(params)
+        rescue AuthenticationError
+          raise if retried
+          retried = true
+          fetch_access_token
+
+          retry
+        end
```

Comment:
I believe we can remove `retry` block if `access_token_near_expiration?` is refactored.

```suggestion
        Request::Api.new(method, path, client).call(params)
```

Reply:
No, Eda wanted the retry because if someone else invalidates the tokens for whatever reason or you store the time incorrectly, you should be able to retry it.